src/shell/shell.c

```diff
@@ -1318,7 +1318,9 @@ int main (int argc, char *argv[])
             else if (errno == ENOENT)
                 ec = 127;
             shell_die (ec, "task %d: start failed: %s: %s",
-                       i, flux_cmd_arg (task->cmd, 0), strerror (errno));
+                       task->rank,
+                       flux_cmd_arg (task->cmd, 0),
+                       strerror (errno));
```

Comment:
If this is where we end up when an executable cannot be started on one of the nodes, it might also be helpful to call out the hostname, if that's not too inconvenient.

e.g. this is what I get when I repeat the test from #4165:
```
 garlick@picl0~$ flux mini run -N4 -n16 stress --timeout 60 --cpu 1
0.794s: flux-shell[0]: FATAL: task 0: start failed: stress: No such file or directory
0.795s: job.exception type=exec severity=0 task 0: start failed: stress: No such file or directory
flux-job: task(s) exited with exit code 127
```

Reply:
Sorry, I should have said I realize that would be an enhancement, not a deficiency of the proposed pr which is fixing a bug.  Feel free to skip handling it here.

Reply:
Since rank doesn't seem to be useful in log messages, maybe RFC 24 "log" events should include the hostname of the shell which generated them (instead of or in addition to the shell rank), and then front end tools could display the hostname for all log messages?

Reply:
Talking offline, it was decided that since the rank->hostname mapping already exists in R, log consumers that want to display hostnames can do so by fetching R first.